warehouse/integrations/testhelper/setup.go

```diff
@@ -308,6 +308,11 @@ func RetrieveRecordsFromWarehouse(
 			switch item := item.(type) {
 			case time.Time:
 				return item.Format(time.RFC3339)
+			case string:
+				if t, err := time.Parse(time.RFC3339Nano, item); err == nil {
+					return t.Format(time.RFC3339)
+				}
+				return item
```

Comment:
This is required because the driver is trying to Parst and because the [redshift driver is formatting it to strin](https://github.com/rudderlabs/sqlconnect-go/blob/main/sqlconnect/internal/redshift/driver/rows.go#L86-L99)g and we are loosing the data type.